processor/transformprocessor/config.go

```diff
@@ -40,35 +40,35 @@ var _ config.Processor = (*Config)(nil)
 func (c *Config) Validate() error {
 	var errors error
 
-	ottlp := ottl.NewParser(
+	ottltrcesp := ottl.NewParser[ottltraces.TransformContext](
 		traces.Functions(),
 		ottltraces.ParsePath,
 		ottltraces.ParseEnum,
 		component.TelemetrySettings{},
 	)
-	_, err := ottlp.ParseStatements(c.Traces.Queries)
+	_, err := ottltrcesp.ParseStatements(c.Traces.Queries)
 	if err != nil {
 		errors = multierr.Append(errors, err)
 	}
 
-	ottlp = ottl.NewParser(
+	ottlmetricsp := ottl.NewParser[ottldatapoints.TransformContext](
 		metrics.Functions(),
 		ottldatapoints.ParsePath,
 		ottldatapoints.ParseEnum,
 		component.TelemetrySettings{},
 	)
-	_, err = ottlp.ParseStatements(c.Metrics.Queries)
+	_, err = ottlmetricsp.ParseStatements(c.Metrics.Queries)
 	if err != nil {
 		errors = multierr.Append(errors, err)
 	}
 
-	ottlp = ottl.NewParser(
+	ottllogsp := ottl.NewParser[ottllogs.TransformContext](
```

Comment:
Maybe a followup change, but instead of a generic NewParser, should contexts provide a Parser specific to their context?  Something like `ottllogs.NewParser(logs.Functions(), component.TelemetrySettings{}`)?

Reply:
I like that more, since 2 arguments are always coming from that package. Will address it in the next PR.

Reply:
Sweet, do you plan to work on that or do we need an issue?

Reply:
Will do after you approve this 🗡️